src/gui/editorwidgets/qgsrelationreferencewidget.cpp

```diff
@@ -906,9 +918,13 @@ void QgsRelationReferenceWidget::filterChanged()
         {
           QMap<QString, QString> filtersAttrs = filters;
           filtersAttrs[fieldName] = QgsExpression::createFieldEqualityExpression( fieldName, txt );
-          QString expression = filtersAttrs.values().join( QStringLiteral( " AND " ) );
-
           QgsAttributeList subset = attrs;
+
+          QString expression = filterExpression;
+          if ( ! filterExpression.isEmpty() && ! filtersAttrs.values().isEmpty() )
+            filterExpression = QStringLiteral( " AND " );
```

Comment:
Is that correct? Should this overwrite the original filterExpression or did you mean to append?
Also should the filters be put into parentheses?

Reply:
You are right, this is wrong. Thanks for noticing.

Reply:
Since all the groupings are `AND` parentheses are not required, but probably more future proof.

Reply:
It was also using the wrong variable, should be good now.
```diff
-            filterExpression = QStringLiteral( " AND " );
+            expression += QStringLiteral( " AND " );
```